### ISSUE TITLE
Aggregate functions cannot be executed in parallel

### ISSUE THREAD

**Issue by yej2021** [ClickHouse/ClickHouse#53901]
**Describe the situation**
The same SQL statement cannot be executed in parallel for aggregate functions on versions 22.3 and 23.3, but can be executed in parallel on version 21.3. What are the possible reasons I would like to know?

```
 2023.08.28 20:35:11.918747 [ 28170 ]<Trace> AggregatingTransform: Aggregating
 2023.08.28 20:35:11.918854 [ 28170 ]<Trace> Aggregator: Aggregation method: keys128
 2023.08.28 20:35:11.957303 [ 28170 ]<Trace> Aggregator: Converting aggregation data to two-level.
 2023.08.28 20:35:25.250542 [ 28170 ]<Debug> AggregatingTransform: Aggregated. 31158147 to 30779169 rows (from 861.73 MiB) in 13.339189175 sec. (2335835.154 rows/sec., 64.60 MiB/sec.)
 2023.08.28 20:35:25.250682 [ 28170 ]<Debug> AggregatingTransform: Aggregated. 0 to 0 rows (from 0.00 B) in 13.339320749 sec. (0.000 rows/sec., 0.00 B/sec.)
 2023.08.28 20:35:25.250729 [ 28170 ]<Debug> AggregatingTransform: Aggregated. 0 to 0 rows (from 0.00 B) in 13.339351751 sec. (0.000 rows/sec., 0.00 B/sec.)
 2023.08.28 20:35:25.250770 [ 28170 ]<Debug> AggregatingTransform: Aggregated. 0 to 0 rows (from 0.00 B) in 13.339378019 sec. (0.000 rows/sec., 0.00 B/sec.)
 2023.08.28 20:35:25.250810 [ 28170 ]<Debug> AggregatingTransform: Aggregated. 0 to 0 rows (from 0.00 B) in 13.339372452 sec. (0.000 rows/sec., 0.00 B/sec.)
 2023.08.28 20:35:25.250850 [ 28170 ]<Debug> AggregatingTransform: Aggregated. 0 to 0 rows (from 0.00 B) in 13.3393918 sec. (0.000 rows/sec., 0.00 B/sec.)
 2023.08.28 20:35:25.250904 [ 28170 ]<Debug> AggregatingTransform: Aggregated. 0 to 0 rows (from 0.00 B) in 13.339421081 sec. (0.000 rows/sec., 0.00 B/sec.)
 2023.08.28 20:35:25.250947 [ 28170 ]<Debug> AggregatingTransform: Aggregated. 0 to 0 rows (from 0.00 B) in 13.339446584 sec. (0.000 rows/sec., 0.00 B/sec.)
 2023.08.28 20:35:25.250987 [ 28170 ]<Debug> AggregatingTransform: Aggregated. 0 to 0 rows (from 0.00 B) in 13.339459291 sec. (0.000 rows/sec., 0.00 B/sec.)
 2023.08.28 20:35:25.251045 [ 28170 ]<Debug> AggregatingTransform: Aggregated. 0 to 0 rows (from 0.00 B) in 13.339501337 sec. (0.000 rows/sec., 0.00 B/sec.)
 2023.08.28 20:35:25.251129 [ 28170 ]<Debug> AggregatingTransform: Aggregated. 0 to 0 rows (from 0.00 B) in 13.339571647 sec. (0.000 rows/sec., 0.00 B/sec.)
 2023.08.28 20:35:25.251210 [ 28170 ]<Debug> AggregatingTransform: Aggregated. 0 to 0 rows (from 0.00 B) in 13.339641359 sec. (0.000 rows/sec., 0.00 B/sec.)
 2023.08.28 20:35:25.251298 [ 28170 ]<Debug> AggregatingTransform: Aggregated. 0 to 0 rows (from 0.00 B) in 13.339707154 sec. (0.000 rows/sec., 0.00 B/sec.)
 2023.08.28 20:35:25.251388 [ 28170 ]<Debug> AggregatingTransform: Aggregated. 0 to 0 rows (from 0.00 B) in 13.339772462 sec. (0.000 rows/sec., 0.00 B/sec.)
 2023.08.28 20:35:25.251492 [ 28170 ]<Debug> AggregatingTransform: Aggregated. 0 to 0 rows (from 0.00 B) in 13.339857772 sec. (0.000 rows/sec., 0.00 B/sec.)
 2023.08.28 20:35:25.251585 [ 28170 ]<Debug> AggregatingTransform: Aggregated. 0 to 0 rows (from 0.00 B) in 13.339930889 sec. (0.000 rows/sec., 0.00 B/sec.)
 2023.08.28 20:35:25.251651 [ 28170 ]<Trace> Aggregator: Merging aggregated data
 2023.08.28 20:35:52.743764 [ 28170 ]<Trace> AggregatingTransform: Aggregating
 2023.08.28 20:35:52.743925 [ 28170 ]<Trace> Aggregator: Aggregation method: keys256
 2023.08.28 20:35:54.850176 [ 28170 ]<Debug> AggregatingTransform: Aggregated. 51413988 to 189489 rows (from 1.39 GiB) in 42.937845166 sec. (1197404.942 rows/sec., 33.12 MiB/sec.)
```
**Expected performance**
Aggregate functions can be executed in parallel， and execution performance can be improved.

**Additional context**
**1、create MV**
```
CREATE MATERIALIZED VIEW IF NOT EXISTS test.mv
TO test.target
AS
SELECT
    id,
    time,
    toStartOfHour(toDateTime(time + time_zone_offset * 60 + dst_offset * 60), 'UTC') as rttime,
    toYear(rttime) as year,
    toMonth(rttime) as month,
    toDayOfMonth(rttime) as day,
    toHour(rttime) as hour,
    toStartOfHour(rttime) as result_time_hour,
    toStartOfDay(rttime) as result_time_day,
    toUInt64(rttime) as local_time,
    local_time - time_zone_offset * 60 - dst_offset * 60 as utc_time,
    time_zone_offset,
    dst_offset,
    maxState(NOW()) as insert_time_state,

    argMaxState(cnt_30014, insert_time) as cnt_30014_state,
    argMaxState(concat(toString(ne_local_time), ':', toString(cnt_30014)), insert_time) as cnt_30014_map_state,
    argMaxState(if(isNotNull(cnt_30016), cnt_30016, null), insert_time) as cnt_30016_valid_state,
    argMaxState(cnt_30018, insert_time) as cnt_30018_state,
    argMaxState(if(cnt_30018 != 0, cnt_30018, null), insert_time) as cnt_30018_valid_state,
    argMaxState(if(isNotNull(cnt_30019), cnt_30019, null), insert_time) as cnt_30019_valid_state,
    argMaxState(if(isNotNull(cnt_30020), cnt_30020, null), insert_time) as cnt_30020_valid_state,
    argMaxState(if(cnt_30022 = 513, cnt_30022, 0), insert_time) as cnt_30022_eq513_state,
    argMaxState(cnt_30026, insert_time) as cnt_30026_state,
    argMaxState(if(cnt_30026 != 0, cnt_30026, null), insert_time) as cnt_30026_valid_state,
    argMaxState(cnt_30028, insert_time) as cnt_30028_state,
    argMaxState(if(isNotNull(cnt_30028), cnt_30028, null), insert_time) as cnt_30028_valid_state,

    argMaxState(v_cnt_deduplicate.cnt_30014, v_cnt_deduplicate.insert_time) as last_cnt_30014_state,
    argMaxState(v_cnt_deduplicate.cnt_30014_map, v_cnt_deduplicate.insert_time) as last_cnt_30014_map_state,
    argMaxState(v_cnt_deduplicate.cnt_30016_valid, v_cnt_deduplicate.insert_time) as last_cnt_30016_valid_state,
    argMaxState(v_cnt_deduplicate.cnt_30018, v_cnt_deduplicate.insert_time) as last_cnt_30018_state,
    argMaxState(v_cnt_deduplicate.cnt_30018_valid, v_cnt_deduplicate.insert_time) as last_cnt_30018_valid_state,
    argMaxState(v_cnt_deduplicate.cnt_30019_valid, v_cnt_deduplicate.insert_time) as last_cnt_30019_valid_state,
    argMaxState(v_cnt_deduplicate.cnt_30020_valid, v_cnt_deduplicate.insert_time) as last_cnt_30020_valid_state,
    argMaxState(v_cnt_deduplicate.cnt_30022_eq513, v_cnt_deduplicate.insert_time) as last_cnt_30022_eq513_state,
    argMaxState(v_cnt_deduplicate.cnt_30026, v_cnt_deduplicate.insert_time) as last_cnt_30026_state,
    argMaxState(v_cnt_deduplicate.cnt_30026_valid, v_cnt_deduplicate.insert_time) as last_cnt_30026_valid_state,
    argMaxState(v_cnt_deduplicate.cnt_30028, v_cnt_deduplicate.insert_time) as last_cnt_30028_state,
    argMaxState(v_cnt_deduplicate.cnt_30028_valid, v_cnt_deduplicate.insert_time) as last_cnt_30028_valid_state,

    maxState(v_cnt_deduplicate.insert_time) as last_insert_time_state
FROM test.base_tab LEFT JOIN (
    SELECT
        id,
        time,
        time_zone_offset,
        dst_offset,
        maxMerge(insert_time_state) as insert_time,
        argMaxMerge(cnt_30014_state) as cnt_30014,
        argMaxMerge(cnt_30014_map_state) as cnt_30014_map,
        argMaxMerge(cnt_30016_valid_state) as cnt_30016_valid,
        argMaxMerge(cnt_30018_state) as cnt_30018,
        argMaxMerge(cnt_30018_valid_state) as cnt_30018_valid,
        argMaxMerge(cnt_30019_valid_state) as cnt_30019_valid,
        argMaxMerge(cnt_30020_valid_state) as cnt_30020_valid,
        argMaxMerge(cnt_30022_eq513_state) as cnt_30022_eq513,
        argMaxMerge(cnt_30026_state) as cnt_30026,
        argMaxMerge(cnt_30026_valid_state) as cnt_30026_valid,
        argMaxMerge(cnt_30028_state) as cnt_30028,
        argMaxMerge(cnt_30028_valid_state) as cnt_30028_valid,

        argMaxMerge(last_cnt_30014_state) as last_cnt_30014,
        argMaxMerge(last_cnt_30014_map_state) as last_cnt_30014_map,
        argMaxMerge(last_cnt_30016_valid_state) as last_cnt_30016_valid,
        argMaxMerge(last_cnt_30018_state) as last_cnt_30018,
        argMaxMerge(last_cnt_30018_valid_state) as last_cnt_30018_valid,
        argMaxMerge(last_cnt_30019_valid_state) as last_cnt_30019_valid,
        argMaxMerge(last_cnt_30020_valid_state) as last_cnt_30020_valid,
        argMaxMerge(last_cnt_30022_eq513_state) as last_cnt_30022_eq513,
        argMaxMerge(last_cnt_30026_state) as last_cnt_30026,
        argMaxMerge(last_cnt_30026_valid_state) as last_cnt_30026_valid,
        argMaxMerge(last_cnt_30028_state) as last_cnt_30028,
        argMaxMerge(last_cnt_30028_valid_state) as last_cnt_30028_valid,

        maxMerge(last_insert_time_state) as last_insert_time
    FROM test.target 
    WHERE id in (SELECT id FROM test.base_tab)
    AND time in (SELECT time FROM test.base_tab)
    GROUP BY id, time, time_zone_offset, dst_offset
    ) v_cnt_deduplicate
ON test.base_tab.id = v_cnt_deduplicate.id
    AND test.base_tab.time = v_cnt_deduplicate.time
GROUP BY id, time, time_zone_offset, dst_offset;
```
**2、insert base_tab data**
```
insert into base_tab select xxxxx;
```

**Comment by UnamedRus** [comment]
It seems, that this issue is about: "Materialized VIEWs are being executed in single thread" and it's expected, i think.

Can you add in issue name that it's about MV?

**Comment by den-crane** [comment]
duplicate of https://github.com/ClickHouse/ClickHouse/issues/53590

**Comment by den-crane** [comment]
duplicate of https://github.com/ClickHouse/ClickHouse/issues/53480
